staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -87,7 +87,7 @@ type Config struct {
 	AuthConfigPersister AuthProviderConfigPersister
 
 	// Exec-based authentication provider.
-	ExecProvider *clientcmdapi.ExecConfig
+	Exec Exec
```

Comment:
can we avoid adding this nesting level? this is an extremely widely used struct and changes here are are disruptive

Reply:
I remember you and @enj talking about this in https://github.com/kubernetes/kubernetes/pull/91192/files#r444297938.

Of the two options you presented in that comment, I think I like `ClusterConfig runtime.Object to ExecConfig` the best, to try to avoid adding another field to `rest.Config`?

Reply:
https://github.com/kubernetes/kubernetes/pull/95489/commits/bd4af0df52b7da2a838795ccf9a8760a0316da48